controllers/reconcilers/grafana/ingress_reconciler.go

```diff
@@ -123,6 +123,22 @@ func (r *IngressReconciler) getIngressAdminURL(ingress *v1.Ingress) string {
 		}
 	}
 
+	// if all fails, try to get access trough the load balancer
+	var loadBalancerIp = ""
+	for _, lb := range ingress.Status.LoadBalancer.Ingress {
+		if lb.Hostname != "" {
+			hostname = lb.Hostname
```

Comment:
I think this part should be rewritten. Say, hostname is defined in one of `ingress.Spec.Rules` and you also have a hostname in `ingress.Status.LoadBalancer.Ingress`. With the current implementation, the latter would take precedence whereas we have to rather rely on the former.
As an option, you could reuse the same variable name for either of the two (`lb.Hostname`, `lb.IP`) and then save the value into `hostname` if `hostname` is still empty by that point.

Reply:
that's a good point. I was adding this code because otherwise I couldn't test my change in k3d. I'll make sure that it's only applied if the hostname couldn't be determined before.